src/controllers/WeaponController.ts

```diff
@@ -16,4 +16,19 @@ export class WeaponController {
     this.weaponProvider.createWeapon(weapon);
     return weapon;
   }
+
+  @Get('') //GET /
+  public getAllWeapons(@Query('element') element: Element, @Query('type') type: Archetype) :Weapon[] {
```

Comment:
The query params can be optional. If you implement the FilterOptions interface below, since those are optional too you don't need any additional logic.
```
const opts: FilterOptions = {
  element,
  type
}
```
Then call the `getWeapons` function in the provider with `this.provider.getWeapons(opts)`.